src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -98,6 +100,16 @@ - (nullable instancetype)initWithAddress:(NSString *)address {
       _address = address;
       _secure = YES;
       kHostCache[address] = self;
+
+      if (!connectivityMonitor) {
+        connectivityMonitor =
+            [GRPCConnectivityMonitor monitorWithHost:hostURL.host];
```

Comment:
Yea. I think the host name does not matters here and assuming all the host names have the same reachability (we talked about this yesterday I remember). I don't need ReachabilityRef to all the hosts. As long as there is at least one host cached, I'm gonna need one ReachabilityRef to flush the channel.